referenceframe/frame.go

```diff
@@ -445,3 +445,96 @@ func (rf rotationalFrame) MarshalJSON() ([]byte, error) {
 
 	return json.Marshal(temp)
 }
+
+type sevenDFrame struct {
+	*baseFrame
+	collisionGeometry spatial.Geometry
+}
+
+// New7DFrame created a frame with 7 degrees of freedom.
```

Comment:
Documenting the order of the Inputs will be important for understanding how to use this.

Also there's going to need to be a bunch of name updates if we change the name